source/postprocess/tracers.cc

```diff
@@ -93,8 +93,12 @@ namespace aspect
           last_output_time = this->get_time() - output_interval;
         }
 
-      // Advance the particles in the world to the current time
-      world.advance_timestep();
+      // Do not advect the particles in the initial refinement stage
+      const bool in_initial_refinement = (this->get_timestep_number() == 0)
+          && (this->get_pre_refinement_step() < this->get_parameters().initial_adaptive_refinement);
+      if (!in_initial_refinement)
+        // Advance the particles in the world to the current time
+        world.advance_timestep();
```

Comment:
I'm confused because I think I understand what this code does, but that doesn't seem to match what you write in your introduction to the patch:
........................
...only writing files once at the first initial adaptive refinement step to not mess up the naming scheme of the output files. Beware that this means for an otherwise unchanged model that setting 'Run postprocessors on initial refinement = true' will change the particle output in the first timestep, because it outputs the state before the adaptive refinement instead of after the initial refinement (as it will also do for the visualization output).
......
The code only seems to run the advancement after the solution has been computed after the *last* of the pre-refinement steps. When you say "only writing files once at the first initial adaptive refinement step" I guess you mean "once" as in "only one time" as opposed to "after a condition has been met", and the "first initial adaptive refinement step" is in fact the "first time step, which may consist of multiple sub-steps separated by refinement"?

Also, why does the output change?